packages/angular_devkit/build_angular/src/utils/webpack-browser-config.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { BuilderContext } from '@angular-devkit/architect/src/index2';
+import {
+  experimental,
+  getSystemPath,
+  logging,
+  normalize,
+  resolve,
+  schema,
+  virtualFs,
+} from '@angular-devkit/core';
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as webpack from 'webpack';
+import { WebpackConfigOptions } from '../angular-cli-files/models/build-options';
+import { readTsconfig } from '../angular-cli-files/utilities/read-tsconfig';
+import { Schema as BrowserBuilderSchema } from '../browser/schema';
+import { NormalizedBrowserBuilderSchema, defaultProgress, normalizeBrowserSchema } from '../utils';
+
+const SpeedMeasurePlugin = require('speed-measure-webpack-plugin');
+const webpackMerge = require('webpack-merge');
+
+type BrowserWebpackConfigOptions = WebpackConfigOptions<NormalizedBrowserBuilderSchema>;
+
+export async function generateWebpackConfig(
+  workspaceRoot: string,
+  projectRoot: string,
+  sourceRoot: string | undefined,
+  options: NormalizedBrowserBuilderSchema,
+  webpackPartialGenerator: (wco: BrowserWebpackConfigOptions) => webpack.Configuration[],
+  logger: logging.LoggerApi,
+): Promise<webpack.Configuration> {
+  // Ensure Build Optimizer is only used with AOT.
+  if (options.buildOptimizer && !options.aot) {
+    throw new Error(`The 'buildOptimizer' option cannot be used without 'aot'.`);
+  }
+
+  const tsConfigPath = path.resolve(workspaceRoot, options.tsConfig);
+  const tsConfig = readTsconfig(tsConfigPath);
+
+  // tslint:disable-next-line:no-implicit-dependencies
+  const projectTs = await import('typescript');
+
+  const supportES2015 = tsConfig.options.target !== projectTs.ScriptTarget.ES3
+    && tsConfig.options.target !== projectTs.ScriptTarget.ES5;
+
+  const wco: BrowserWebpackConfigOptions = {
+    root: workspaceRoot,
+    logger: logger.createChild('webpackConfigOptions'),
+    projectRoot,
+    sourceRoot,
+    buildOptions: options,
+    tsConfig,
+    tsConfigPath,
+    supportES2015,
+  };
+
+  wco.buildOptions.progress = defaultProgress(wco.buildOptions.progress);
+
+  const partials = webpackPartialGenerator(wco);
+  const webpackConfig = webpackMerge(partials);
+
+  if (options.profile || process.env['NG_BUILD_PROFILING']) {
+    const smp = new SpeedMeasurePlugin({
+      outputFormat: 'json',
+      outputTarget: path.resolve(workspaceRoot, 'speed-measure-plugin.json'),
+    });
+
+    return smp.wrap(webpackConfig);
+  }
+
+  return webpackConfig;
+}
+
+
+export async function generateBrowserWebpackConfigFromWorkspace(
```

Comment:
Should this be merged with `buildBrowserWebpackConfigFromWorkspace` from the browser builder and exported?